internal/storage/service.go

```diff
@@ -238,6 +238,13 @@ func (s *serviceImpl) ListLocations(
 	return s.storagePersistence.ListLocations(ctx)
 }
 
+func (s *serviceImpl) ListAips(
+	ctx context.Context,
+	payload *goastorage.ListAipsPayload,
+) (*goastorage.AIPs, error) {
+	return s.storagePersistence.ListAIPs(ctx, payload)
+}
+
```

Comment:
This service works a bit different than the ingest service. In here, we trust more on the generated Goa types and even return those errors from the persistence layer, so this method just goes directly to the persistence layer.

Reply:
Which raises the question, why do we need an ent client and a service when the service could be the ent client at the same time. If we impose that the interactions with storage are done through the service (not directly going to the persistence layer), I think that could simplify the current implementation.

Reply:
Ideally all of the parameters and returns from this service layer should be agnostic to the API implementation (i.e. not accepting `*goastorage.ListAipsPayload` and returning `*goastorage.AIPs`) and the persistence client implementation, like we've done in the ingest service.  Adapters should be used in between layers to convert types.

Reply:
I agree, but I also remember Sevein raising good points about this approach (API first, etc.), I'd like to check with him first before changing the entire service.

Reply:
Oh man, I hope "API first" doesn't imply that we should be tightly coupling our application code to the API implementation. :grimacing: